### PR TITLE
docs(list): add api docs #61

### LINKED ISSUE #61: Add API docs to List component
## Result  
The list components lacks public API documentation.

## Expected result  
The component should have its public API documented.

### PR DESCRIPTION
**Pull request by zhosti**
Closes #61

### CODE REVIEW

**Review by igdmdimitrov — APPROVED** [review]

File: src/components/list/list-header.ts [hunk 1]
```diff
@@ -1,6 +1,13 @@
 import { html, LitElement } from 'lit';
 import { styles } from './list-header.material.css';
 
+/**
+ * Header component for list items.
```
Comment (gedinakova) [inline src/components/list/list-header.ts:5]: Isn't this just a header list item. "Header component for list items" kind of has a different meaning.

File: src/components/list/list-item.ts [hunk 2]
```diff
@@ -1,6 +1,25 @@
 import { html, LitElement } from 'lit';
 import { styles } from './list-item.material.css';
 
+/**
+ * The list-item component is a container
+ * intended for row items in the list component.
+ *
+ * @element igc-list-item
+ *
+ * @slot - Renders a custom content.
```
Comment (gedinakova) [inline src/components/list/list-item.ts:10]: No 'a' before content.

File: src/components/list/list-item.ts [hunk 3]
```diff
@@ -1,6 +1,25 @@
...
+ * @slot start - Renders the content before the title and subtitle.
```
Comment (gedinakova) [inline src/components/list/list-item.ts:11]: No 'the' before content.

File: src/components/list/list-item.ts [hunk 4]
```diff
@@ -1,6 +1,25 @@
...
+ * @slot end - Renders the content after the title and subtitle.
```
Comment (gedinakova) [inline src/components/list/list-item.ts:12]: No 'the' before content. It rather "Renders content after all other content."

File: src/components/list/list-item.ts [hunk 5]
```diff
@@ -1,6 +1,25 @@
...
+ * @slot title - Renders the title.
+ * @slot subtitle - Renders the subtitle.
+ *
+ * @csspart start - The start container.
+ * @csspart end - The end container.
+ * @csspart content - The header and custom template container.
```
Comment (gedinakova) [inline src/components/list/list-item.ts:18]: Probably "content" instead of "template"?

File: src/components/list/list-item.ts [hunk 6]
```diff
@@ -1,6 +1,25 @@
...
+ * @csspart header - The title and subtitle container.
+ * @csspart title - The title wrapper.
```
Comment (gedinakova) [inline src/components/list/list-item.ts:20]: "Container" instead of "wrapper".

File: src/components/list/list-item.ts [hunk 7]
```diff
@@ -1,6 +1,25 @@
...
+ * @csspart subtitle - The subtitle wrapper.
```
Comment (gedinakova) [inline src/components/list/list-item.ts:21]: Same as above.

**Review by gedinakova — CHANGES_REQUESTED** [review]

File: src/components/list/list-item.ts [hunk 8]
```diff
@@ -1,6 +1,25 @@
 import { html, LitElement } from 'lit';
 import { styles } from './list-item.material.css';
 
+/**
+ * The list-item component is a container
+ * intended for row items in the list component.
```
Comment (onlyexeption) [inline src/components/list/list-item.ts:6]: I think intended is unnecessary here

**Review by onlyexeption — DISMISSED** [review]

File: src/components/list/list-header.ts [hunk 9]
```diff
@@ -1,6 +1,13 @@
 import { html, LitElement } from 'lit';
 import { styles } from './list-header.material.css';
 
+/**
+ * Header list item.
+ *
+ * @element igc-list-header
+ *
+ * @slot - Renders a header for the list items.
```
Comment (gedinakova) [inline src/components/list/list-header.ts:9]: Let's change this to "Renders header list item's content.".

**Review by gedinakova — CHANGES_REQUESTED** [review]

**Review by gedinakova — APPROVED** [review]

**Pull request merged**